src/drawers/drawer.ts

```diff
@@ -5,238 +5,101 @@
-    step.animator.animate(selection, step.attrToAppliedProjector);
-    if (this._className != null) {
-      this.selection().classed(this._className, true);
+  public useSVG(parent: d3.Selection<SVGElement, any, any, any>) {
+    if (this._currentDrawer != null) {
+      this._currentDrawer.remove();
     }
-  }
-
-  protected _drawStepCanvas(data: any[], step: Drawers.AppliedDrawStep) {
-    Utils.Window.warn("canvas rendering not yet implemented on " + (<any> this.constructor).name);
-  }
-
-  private _appliedProjectors(attrToProjector: AttributeToProjector): AttributeToAppliedProjector {
-    const modifiedAttrToProjector: AttributeToAppliedProjector = {};
-    Object.keys(attrToProjector).forEach((attr: string) => {
-      modifiedAttrToProjector[attr] =
-        (datum: any, index: number) => attrToProjector[attr](datum, index, this._dataset);
-    });
-
-    return modifiedAttrToProjector;
+    const svgDrawer = this._svgDrawerFactory();
+    svgDrawer.attachTo(parent);
+    this._currentDrawer = svgDrawer;
   }
 
   /**
-   * Calculates the total time it takes to use the input drawSteps to draw the input data
-   *
-   * @param {any[]} data The data that would have been drawn
-   * @param {Drawers.DrawStep[]} drawSteps The DrawSteps to use
-   * @returns {number} The total time it takes to draw
+   * Remove the old drawer and use Canvas rendering from now on.
    */
-  public totalDrawTime(data: any[], drawSteps: Drawers.DrawStep[]) {
-    let delay = 0;
-    drawSteps.forEach((drawStep, i) => {
-      delay += drawStep.animator.totalTime(data.length);
-    });
-
-    return delay;
-  }
-
-  /**
-   * Draws the data into the renderArea using the spefic steps and metadata
-   *
-   * @param{any[]} data The data to be drawn
-   * @param{DrawStep[]} drawSteps The list of steps, which needs to be drawn
-   */
-  public draw(data: any[], drawSteps: Drawers.DrawStep[]) {
-    const appliedDrawSteps: Drawers.AppliedDrawStep[] = drawSteps.map((dr: Drawers.DrawStep) => {
-      const attrToAppliedProjector = this._appliedProjectors(dr.attrToProjector);
-      return {
-        attrToAppliedProjector: attrToAppliedProjector,
-        animator: dr.animator,
-      };
-    });
-
-    if (this._renderArea != null) {
-      this._bindSelectionData(data);
-      this._cachedSelectionValid = false;
-
-      let delay = 0;
-      appliedDrawSteps.forEach((drawStep, i) => {
-        Utils.Window.setTimeout(() => this._drawStep(drawStep), delay);
-        delay += drawStep.animator.totalTime(data.length);
-      });
-    } else if (this._canvas != null) {
-      // don't support animations for now; just draw the last draw step immediately
-      const lastDrawStep = appliedDrawSteps[appliedDrawSteps.length - 1];
-      Utils.Window.setTimeout(() => this._drawStepCanvas(data, lastDrawStep), 0);
-    } else {
-      throw new Error("Drawer's canvas and renderArea are both null!");
+  public useCanvas(canvas: d3.Selection<HTMLCanvasElement, any, any, any>) {
```

Comment:
maybe it'd still be possible share one canvas across drawers without keeping track of it in the component, but this should be fine for now